src/Tools/dotnet-monitor/Exceptions/ExceptionsService.cs

```diff
@@ -48,6 +55,22 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                 {
                     // Get default process
                     IProcessInfo pi = await _diagnosticServices.GetProcessAsync(processKey: null, stoppingToken);
+
+                    // If previous checked configuration and it did not pass, do not check again or attempt
+                    // to start the event pipe session.
+                    UniqueProcessKey key = new(pi.EndpointInfo.ProcessId, pi.EndpointInfo.RuntimeInstanceCookie);
+                    if (_unconfiguredProcesses.Contains(key))
+                    {
+                        throw new MonitoringException(string.Empty);
+                    }
+
+                    // Validate that the process is configured correctly for collecting exceptions.
+                    if (!await _startupHookValidator.CheckAsync(pi.EndpointInfo, stoppingToken))
+                    {
+                        _unconfiguredProcesses.Add(key);
+                        throw new MonitoringException(string.Empty);
```

Comment:
I'm guessing there's no message here since you already log from CheckAsync?

Reply:
Correct, and the exception is not user visible.